BASE/src/Microsoft.ApplicationInsights/Extensibility/Implementation/Platform/PlatformImplementation.cs

```diff
@@ -119,6 +120,48 @@ public string GetMachineName()
             return this.hostName ?? (this.hostName = GetHostName());
         }
 
+        public void TestDirectoryPermissions(DirectoryInfo directory)
```

Comment:
`IPlatform` is the interface this class implements. `TestDirectoryPermissions` is not part of it. Don't think it should be part of it.

Reply:
I want this method in a common location so I can reuse it.
I added it to the interface. I thought that interacting with the file system would be appropriate to put in the IPlatform. 
If you disagree, I can move it to a helper class instead. :)